packages/pages-shared/asset-server/handler.ts

```diff
@@ -732,6 +744,26 @@ function isPreview(url: URL): boolean {
 	return false;
 }
 
+/** Checks if a response is older than CACHE_PRESERVATION_WRITE_FREQUENCY
+ * and should be written to cache again to reset it's expiration.
+ */
+export function isPreservationCacheResponseExpiring(
```

Comment:
I did not do the same thing for the Early Hints cache because it's set to 30 days and already has the entire cache expire on every deployment anyway.